src/persistence/profile.cpp

```diff
@@ -520,12 +525,19 @@ void Profile::setPassword(QString newPassword)
 {
     QList<HistoryKeeper::HistMessage> oldMessages = HistoryKeeper::exportMessagesDeleteFile();
     QByteArray avatar = loadAvatarData(core->getSelfId().publicKey);
-
+    QString oldPassword = password;
     password = newPassword;
     passkey = *core->createPasskey(password);
     saveToxSave();
-
     HistoryKeeper::getInstance()->importMessages(oldMessages);
     Nexus::getDesktopGUI()->reloadHistory();
     saveAvatar(avatar, core->getSelfId().publicKey);
+
+    QVector<uint32_t> friendList = core->getFriendList();
+    QVectorIterator<uint32_t> i(friendList);
+    while (i.hasNext())
+    {
+        QString friendPublicKey = core->getFriendPublicKey(i.next());
+        saveAvatar(loadAvatarData(friendPublicKey,oldPassword),friendPublicKey);
```

Comment:
@tux3  So I just looked this over again, and when loadAvatarData is called, it is called with the password (even if that password was formerly empty). When this happens, loadAvatarData will call avatarPath once, which will generate the new path, but right afterwards, loadAvatarData checks if the password (the one which was passed) is empty. If it is, the path is changed to the result of the non-hashed avatarPath. Have I missed something?